tck/src/test/java/org/reactivestreams/tck/PublisherVerificationTest.java

```diff
@@ -717,6 +735,79 @@ public void cancel() {
   }
 
   /**
+   * Verification using a Publisher that supports multiple subscribers
+   * @param shouldBlowUp if true {@link RuntimeException} will be thrown during second subscription.
+   */
+  final PublisherVerification<Integer> multiSubscribersPublisherVerification(final boolean shouldBlowUp) {
+    return new PublisherVerification<Integer>(newTestEnvironment()) {
+
+      @Override
+      public Publisher<Integer> createPublisher(final long elements) {
+        return new Publisher<Integer>() {
+
+          private final Collection<Demand> demands = new CopyOnWriteArrayList<Demand>();
+          private final AtomicLong source = new AtomicLong(elements);
+
+          @Override
+          public void subscribe(Subscriber<? super Integer> s) {
+            if (shouldBlowUp && !demands.isEmpty()) {
+              // onSubscribe first
+              new Demand(s);
+              s.onError(new RuntimeException("Unexpected additional subscriber"));
+            } else {
+              demands.add(new Demand(s));
+            }
+          }
+
+          class Demand implements Subscription {
```

Comment:
Let's not call it demand, don't want to mislead people into getting to this class by "open class by name" accidentally and looking for "what's that demand tha these reactive streams people keep taking about" when developing an impl attempting to pass the TCK. Some boring name here would likely be better..